src/Groups/MatGrp.jl

```diff
@@ -0,0 +1,888 @@
+      G.ring_iso=fm.fr
+      G.mat_iso=fm
+
+   elseif sym === :X
+      if isdefined(G,:descr)
+         if !isdefined(G,:ring_iso)
+            fm = gen_mat_iso(G.deg, G.ring)
+            G.ring_iso=fm.fr
+            G.mat_iso=fm
+         end
+         assign_from_description(G)
+      elseif isdefined(G,:gens)
+         if !isdefined(G,:ring_iso)
+            fm = gen_mat_iso(G.deg, G.ring)
+            G.ring_iso=fm.fr
+            G.mat_iso=fm
+         end
+         V = GAP.julia_to_gap([g.X for g in gens(G)])
+         G.X=GAP.Globals.Group(V)
+      else
+         error("Cannot determine underlying GAP object")
+      end
+   end
+
+   return getfield(G, sym)
+
+end
+
+
+function Base.getproperty(x::MatrixGroupElem, sym::Symbol)
+
+   if isdefined(x,sym) return getfield(x,sym) end
+
+   if sym === :X
+      x.X = x.parent.mat_iso(x.elm)
+   elseif sym == :elm
+      x.elm = x.parent.mat_iso(x.X)
+   end
+   return getfield(x,sym)
+end
+
+Base.IteratorSize(::Type{<:MatrixGroup}) = Base.SizeUnknown()
+
+function Base.iterate(G::MatrixGroup)
+  L=GAP.Globals.Iterator(G.X)
+  @assert ! GAP.Globals.IsDoneIterator(L)
+  i = GAP.Globals.NextIterator(L)
+  return MatrixGroupElem(G, G.mat_iso(i),i), L
+end
+
+function Base.iterate(G::MatrixGroup, state)
+  if GAP.Globals.IsDoneIterator(state)
+    return nothing
+  end
+  i = GAP.Globals.NextIterator(state)
+  return MatrixGroupElem(G, G.mat_iso(i),i), state
+end
+
+########################################################################
+#
+# Membership
+#
+########################################################################
+
+
+function ==(G::MatrixGroup,H::MatrixGroup)
+   G.deg==H.deg || return false
+   G.ring==H.ring || return false
+   if isdefined(G, :descr) && isdefined(H, :descr)
+      return G.descr == H.descr
+   end
+   if isdefined(G, :gens) && isdefined(H, :gens)
+      gens(G)==gens(H) && return true
+   end
+   return G.X==H.X
+end
+
+
+# this saves the value of x.X
+# x_gap = x.X if this is already known, x_gap = nothing otherwise
+function lies_in(x::MatElem, G::MatrixGroup, x_gap)
+   if base_ring(x)!=G.ring || nrows(x)!=G.deg return false, x_gap end
+   if isone(x) return true, x_gap end
+   if isdefined(G,:gens)
+      for g in gens(G)
+         if x==g.elm
+            return true, x_gap
+         end
+      end
+   end
+   if isdefined(G,:descr) && G.descr==:GL
+      if det(x)!=0
+         return true, x_gap
+      else
+         return false, x_gap
+      end
+   elseif isdefined(G,:descr) && G.descr==:SL
+      if det(x)==1
+         return true, x_gap
+      else
+         return false, x_gap
+      end
+   else
+      if x_gap==Nothing x_gap = G.mat_iso(x) end
+     # x_gap !=nothing || x_gap = G.mat_iso(x)
+      return GAP.Globals.in(x_gap,G.X), x_gap
+   end
+end
+
+Base.in(x::MatElem, G::MatrixGroup) = lies_in(x,G,Nothing)[1]
+
+function Base.in(x::MatrixGroupElem, G::MatrixGroup)
+   isdefined(x,:X) && lies_in(x.elm,G,x.X)[1]
```

Comment:
Do you perhaps means this? Otherwise I don't understand the code at all
```suggestion
   isdefined(x,:X) && return lies_in(x.elm,G,x.X)[1]
```

Reply:
Yes, of course :-)